core/embed/rust/src/ui/model_mercury/component/number_input_slider.rs

```diff
@@ -0,0 +1,204 @@
+use crate::ui::{
+    component::{base::ComponentExt, Child, Component, Event, EventCtx},
+    constant::screen,
+    display,
+    event::TouchEvent,
+    geometry::{Grid, Insets, Point, Rect},
+    shape::{self, Renderer},
+};
+
+use super::{theme, Button, ButtonMsg};
+
+pub enum NumberInputSliderDialogMsg {
+    Changed(u16),
+    Confirmed,
+    Cancelled,
+}
+
+pub struct NumberInputSliderDialog {
```

Comment:
Note: copy of `model_t` version. Should be reworked for `mercury` when design done.
Linking it here for now: https://github.com/trezor/trezor-firmware/issues/3748